jenkins/coldlaunch.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash -x
+
+# receive list of apps to run coldlaunch test against
+# i.e. 'calendar clock communications@dialer email'
+APP_LIST=$1
+
+post_to_treeherder() {
+  local RAPTOR_BUILD_STATE=$1
+  local RAPTOR_APP_NAME=$2
+  local TEST_START_TIME=$3
+  local TEST_FINISH_TIME=$4
+  local TEST_FAILURE=$5
+
+  python --version
+  source raptor-env/bin/activate
+  python --version
+
+  post-to-treeherder/submit-to-treeherder.py \
+    --repository b2g-inbound \
+    --build-state ${RAPTOR_BUILD_STATE} \
+    --treeherder-url https://treeherder.allizom.org/ \
+    --treeherder-client-id raptor \
+    --treeherder-secret ${RAPTOR_TREEHERDER_SECRET} \
+    --test-type cold-launch \
+    --app-name ${RAPTOR_APP_NAME} \
+    --start-time ${TEST_START_TIME} \
+    --finish-time ${TEST_FINISH_TIME} \
+    --test-failure ${TEST_FAILURE}
+}
+
+run_coldlaunch_test() {
+  local APP_ORIGIN=$1
+  local ENTRY_POINT=$2
+
+  APP_FAILURE=0
+
+  local FILE_FORMAT="${APP_ORIGIN}"
+  local ENTRY_FLAG=""
+
+  # run coldlaunch
+  if [ -n "$ENTRY_POINT" ] ; then
+    local FILE_FORMAT="${APP_ORIGIN}-${ENTRY_POINT}"
+    local ENTRY_FLAG="--entry-point ${ENTRY_POINT}"
+  fi
+
+  if [ -n "$RAPTOR_DATABASE_NAME" ] ; then
+    local DATABASE_FLAG="--database ${RAPTOR_DATABASE_NAME}"
+  fi
+
+  DEBUG=* node_modules/.bin/raptor test coldlaunch \
+    --serial ${ANDROID_SERIAL} \
+    --app ${APP_ORIGIN} \
+    --runs ${RUNS} \
+    --timeout ${TIMEOUT} \
+    --time ${TEST_TIME} \
+    --metrics ${FILE_FORMAT}.ldjson \
+    --logcat ${FILE_FORMAT}.logcat \
+    ${ENTRY_FLAG} 2>&1 | tee ${FILE_FORMAT}.log
+
```

Comment:
\o/

Raptor fails:

Thu, 07 Jan 2016 21:47:05 GMT raptor:cli Loading plugin raptor-submit@1.0.0
Thu, 07 Jan 2016 21:47:05 GMT raptor:cli Loading plugin raptor-track@2.0.0
Thu, 07 Jan 2016 21:47:05 GMT raptor:cli Loading plugin raptor-bug@2.0.0
[Cold Launch: clock.gaiamobile.org] Preparing to start testing...
[Cold Launch: clock.gaiamobile.org] Testing complete
[Cold Launch: clock.gaiamobile.org] Aborted due to error:

Error: Test timeout exceeded truems
    at ColdLaunch.Phase.timeoutError (/home/rwood/test/dist/es5/lib/phases/phase.js:82:22)
    at null._onTimeout (/home/rwood/test/dist/es5/lib/phases/phase.js:103:18)
    at Timer.listOnTimeout (timers.js:92:15)
+ APP_FAILURE=1

Raptor pass:
+ raptor version
5.0.0
+ APP_FAILURE=0